src/tribler-core/tribler_core/modules/metadata_store/community/gigachannel_community.py

```diff
@@ -1,342 +1,238 @@
 
-        if self.notifier and search_results:
-            self.notifier.notify(
-                NTFY.REMOTE_QUERY_RESULTS, {"uuid": search_request_cache.uuid, "results": search_results}
-            )
+        # ACHTUNG! the returned request cache can be either a SelectRequest or PushbackWindow
+        request = self.request_cache.get(hexlify(peer.mid), response_payload.id)
+        if request is None:
+            self.logger.info(f"No request for response")
+            return
 
-        # Send the updated metadata if any to the responding peer
-        self.respond_with_updated_metadata(peer, reply_blob)
+        # Check for limit on the number of packets per request
+        if request.packets_limit > 1:
+            request.packets_limit -= 1
+        else:
+            self.request_cache.pop(hexlify(peer.mid), response_payload.id)
+
+        processing_results = await self.mds.process_compressed_mdblob_threaded(response_payload.raw_blob)
+        self.logger.info(f"Response result: {processing_results}")
+
+        # Query back the sender for preview contents for the new channels
+        if self.settings.channel_query_back_enabled:
+            new_channels = [md for md, result in processing_results if result in (UNKNOWN_CHANNEL, UNKNOWN_COLLECTION)]
+            for channel in new_channels:
+                request_dict = {
+                    # FIXME: This is a dirty hack, since this is exploitable
+                    "origin_id": channel.id_,
+                    "first": 0,
+                    "last": self.settings.max_channel_query_back,
+                }
+                self.legacy_send_remote_select(peer=peer, **request_dict)
+
+        if isinstance(request, SelectRequest) and request.processing_callback:
+            request.processing_callback(request, processing_results)
+
+    def introduction_response_callback(self, peer, dist, payload):
+        if peer.address in self.network.blacklist or peer.mid in self.queried_peers:
+            return
+        if len(self.queried_peers) >= self.settings.queried_peers_limit:
+            self.new_style_peers.clear()
+            self.queried_peers.clear()
+        self.queried_peers.add(peer.mid)
+
+        if not payload.extra_bytes:
+            # We were introduced to a legacy peer, so send old-style channel request to it
+            self.legacy_send_remote_select_subscribed_channels(peer)
+        elif payload.extra_bytes == MAGIC_GIGACHAN_VERSION_MARK:
+            self.send_remote_select_subscribed_channels(peer)
+            self.new_style_peers.add(peer)
+
+    def get_random_peers(self, sample_size=None):
+        # Randomly sample sample_size peers from the complete list of our peers excluding legacy peers
+        all_peers = set(self.get_peers()).intersection(self.new_style_peers)
+        if sample_size is not None and sample_size < len(all_peers):
+            return sample(all_peers, sample_size)
+        print(len(all_peers))
```

Comment:
I think you forgot to remove this. Not sure why the print-statement checker is not pointing this out.

Reply:
Oops...